3_etl_code/ETL_Orchestration/sql/etl_stem_medical_events.sql

```diff
@@ -282,10 +370,17 @@ LEFT JOIN ( SELECT SOURCE,
                    code,
                    omop_concept_id
             FROM @schema_table_codes_info ) AS fgc
-ON ssfgc.vocabulary_id = fgc.vocabulary_id AND
-   ssfgc.FG_CODE1 IS NOT DISTINCT FROM fgc.FG_CODE1 AND
-   ssfgc.FG_CODE2 IS NOT DISTINCT FROM fgc.FG_CODE2 AND
-   ssfgc.FG_CODE3 IS NOT DISTINCT FROM fgc.FG_CODE3
+ON CASE
+        WHEN ssfgc.SOURCE IN ("BMI","HEIGHT") THEN ssfgc.SOURCE = fgc.SOURCE AND
```

Comment:
i guess you did this bcs the codes are not in service_sector_fg_codes
but I think we have to create them there 
and remove this code

Reply:
Ah no Javier. The codes are in `service_sector_fg_codes` but the problem is the following SQL block
```
ssfgc.vocabulary_id = fgc.vocabulary_id AND
ssfgc.FG_CODE1 IS NOT DISTINCT FROM fgc.FG_CODE1 AND
ssfgc.FG_CODE2 IS NOT DISTINCT FROM fgc.FG_CODE2 AND
ssfgc.FG_CODE3 IS NOT DISTINCT FROM fgc.FG_CODE3
```
Here we are using `vocabulary_id` and if we do this then for the following BMI case

|SOURCE|FG_CODE1|FG_CODE2|FG_CODE3|VOCABULARY_ID|
|:----|:----|:----|:----|:----|
|BMI|NULL|NULL|NULL|FGVisitType|

This will map to every other code with no mapping an example below

|SOURCE|FG_CODE1|FG_CODE2|FG_CODE3|VOCABULARY_ID|FG_CODE5|FG_CODE|
|:----|:----|:----|:----|:----|:----|:----|
|INPAT|NULL|NULL|NULL|FGVisitType|T10|NULL|
|OUTPAT|NULL|NULL|NULL|FGVisitType|T25|NULL|
|PRIM_OUT|NULL|NULL|NULL|FGVisitType|NULL|T20|

Which is why I had to change the above code for BMI and Height codes to the following

```
ssfgc.SOURCE = fgc.SOURCE AND
ssfgc.vocabulary_id = fgc.vocabulary_id AND
ssfgc.FG_CODE1 IS NOT DISTINCT FROM fgc.FG_CODE1 AND
ssfgc.FG_CODE2 IS NOT DISTINCT FROM fgc.FG_CODE2 AND
ssfgc.FG_CODE3 IS NOT DISTINCT FROM fgc.FG_CODE3
```